@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -124,11 +128,83 @@ export class InterpolatedClipPlanesEvaluator implements ClipPlanesEvaluator {
             assert(false, "Unsuported projection type");
         }
 
-        const viewRanges: ViewRanges = { near: nearPlane, far: farPlane, maximum: farPlane };
+        const viewRanges: ViewRanges = {
+            near: nearPlane,
+            far: farPlane,
+            minimum: this.nearMin,
+            maximum: farPlane
+        };
         return viewRanges;
     }
 }
 
+/**
+ * Add support for elevation constraints.
+ */
+export abstract class ElevationBasedClipPlanesEvaluator implements ClipPlanesEvaluator {
```

Comment:
I have the impression we are overengineering this stuff a bit. Why do we need so many evaluators. I think there should be one that solves everything and if the user needs to customize he/she could provide it's own.
Also be aware that this is all public API that we have to maintain. The only thing that should actually be in our API is the ClippingPlanesEvaluator base interface.

Reply:
As you wish, but then lots of work goes to trash... Keep in mind that for globe projection AltitudeBased (top view) works just fine and for planar projections you may also use this one for all 2d like (top-view) maps - faster and easier to understand. The later ones are:
- Interpolated - kept only to have backward compatibility until everything is done (without it we should fix all with single commit - impossible + no body would review that)
- Fixed - that was there always, but hardcoded into MapView / demo app.
- TiltBased - this just adds tilt support to AltitudeBased.

Reply:
What goes to trash? You just merge everything into one class.
From the user point of view it just makes no sense to differentiate for example between tilt based and non tilt based evaluator. If the user is tilting it should just work.

We can maybe move the fixed one out and make a dedicated example that shows how to customize the near and far plane, but I doubt that there is big interest in this if our internal implementation is serving all the needs. We also have to check if anybody uses this in map-view-demo. If yes than we might move it back there and keep coresdk clean.

Reply:
I mean that if we remove entirely AltitudeBasedClipPlanesEvaluator (TopView after refactor) then we resign from still simple and fast evaluator that may be successfully used in any 2D map, or top-view situation.
Keep in mind that is works not only for planar projections, i.e. all globe examples should use it now (see DATASOURCE: features / polygons example), cause we don't have a tilt there. Moreover it is much easier to debug/test new features that are view range dependent, thus I would strongly recommend to keep it for a while (at least).

Reply:
So we have it mainly for performance optimizations? Since we only do the near/far plane evaluation once per frame it is not that critical and we should not do premature-optimzations and we should for sure not put this burden on the user to select the proper `ClipPlanesEvaluator`. If we really need to optimize for performance we should do it under the hood (i.e. have some `if( tilt !== 0)`)

Reply:
Nino just to keep you informed, by default MapViewDemo app will still use InterpolatedEvaluator (legacy/old one). We have special GUI for switching evaluators, this is already merged into master. In order to test (and compare) new evaluators you need to use MapControls->Frustum->PlanesEvaluator combo. If everything is fine and integrates with globe controls with tilt, we will switch default setup for new evaluator.